###Screening variants match to further analysis

Only select those variants with C to T change or G to A change, since our base-editing tools can only change C to T. As the percentage of methylated nucleotides are low in coding region, we first look into those variant that have GC dinucleotide in the mutated point.

In [1]:
!pwd

/home/chengchen/data/base-editing


screen clinvar data from human for C to T transition variant

In [3]:
!ls

clinvar_20181202.vcf  findingVariants.ipynb	 human_orthos_2018.12.06.txt
data_analysis.sh      GC change information.odt  methylation


total number of C to T variant change in clinvar dataset

In [4]:
!awk '$4=="C" && $5=="T" {print $0}' clinvar_20181202.vcf | wc -l

89225


total number of G to A variant change in clinvar dataset

In [5]:
!awk '$4=="G" && $5=="A" {print $0}' clinvar_20181202.vcf | wc -l

88903


select variants and build a new dataset that only store the C (G) to T (A) variants.

In [6]:
!awk '$4=="C" && $5=="T" {print $0}' clinvar_20181202.vcf > humanCtoTvariant.vcf

In [7]:
!awk '$4=="G" && $5=="A" {print $0}' clinvar_20181202.vcf >> humanCtoTvariant.vcf

In [9]:
!head -3 humanCtoTvariant.vcf

1	1014122	542074	C	T	.	.	ALLELEID=514926;CLNDISDB=MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563;CLNDN=Immunodeficiency_38_with_basal_ganglia_calcification;CLNHGVS=NC_000001.11:g.1014122C>T;CLNREVSTAT=criteria_provided,_single_submitter;CLNSIG=Uncertain_significance;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;GENEINFO=ISG15:9636;MC=SO:0001583|missense_variant;ORIGIN=1;RS=150861311
1	1014143	183381	C	T	.	.	ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563;CLNDN=Immunodeficiency_38_with_basal_ganglia_calcification;CLNHGVS=NC_000001.11:g.1014143C>T;CLNREVSTAT=no_assertion_criteria_provided;CLNSIG=Pathogenic;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;CLNVI=OMIM_Allelic_Variant:147571.0003;GENEINFO=ISG15:9636;MC=SO:0001587|nonsense;ORIGIN=1;RS=786201005
1	1014179	542075	C	T	.	.	ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563;CLNDN=Immunodeficiency_38_with_basal_ganglia_calcification;CLNHGVS=NC_000001.11:g.1014179C>T;CLNREVST

In [10]:
!grep -E 'missense|nonsense' humanCtoTvariant.vcf | wc -l

92491


save missense and nonsense variant in humanCtoTvariant.vcf to a new file humanCtoTvariant_coding.vcf

In [11]:
!grep -E 'missense|nonsense' humanCtoTvariant.vcf > humanCtoTvariant_coding.vcf

In [12]:
!head -3 humanCtoTvariant_coding.vcf

1	1014122	542074	C	T	.	.	ALLELEID=514926;CLNDISDB=MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563;CLNDN=Immunodeficiency_38_with_basal_ganglia_calcification;CLNHGVS=NC_000001.11:g.1014122C>T;CLNREVSTAT=criteria_provided,_single_submitter;CLNSIG=Uncertain_significance;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;GENEINFO=ISG15:9636;MC=SO:0001583|missense_variant;ORIGIN=1;RS=150861311
1	1014143	183381	C	T	.	.	ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563;CLNDN=Immunodeficiency_38_with_basal_ganglia_calcification;CLNHGVS=NC_000001.11:g.1014143C>T;CLNREVSTAT=no_assertion_criteria_provided;CLNSIG=Pathogenic;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;CLNVI=OMIM_Allelic_Variant:147571.0003;GENEINFO=ISG15:9636;MC=SO:0001587|nonsense;ORIGIN=1;RS=786201005
1	1014179	542075	C	T	.	.	ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563;CLNDN=Immunodeficiency_38_with_basal_ganglia_calcification;CLNHGVS=NC_000001.11:g.1014179C>T;CLNREVST

Get human fish homologue genes, human_orthos_2018.12.06.txt was download from ZFIN database

In [20]:
!head -4 human_orthos_2018.12.06.txt

Date: 2018.12.06
ZFIN ID	ZFIN Symbol	ZFIN Name	Human Symbol	Human Name	OMIM ID	Gene ID	HGNC ID	Evidence	Pub ID	
ZDB-GENE-000112-47	ppardb	peroxisome proliferator-activated receptor delta b	PPARD	peroxisome proliferator activated receptor delta	600409	5467	9235	AA	ZDB-PUB-060313-16
ZDB-GENE-000112-47	ppardb	peroxisome proliferator-activated receptor delta b	PPARD	peroxisome proliferator activated receptor delta	600409	5467	9235	AA	ZDB-PUB-070210-39


In [13]:
!awk -F "\t" '{print $4}' human_orthos_2018.12.06.txt | sed '1,2d' | sort | uniq| wc -l

13187


In [14]:
!awk -F "\t" '{print $4}' human_orthos_2018.12.06.txt | sed '1,2d' | sort | uniq| head

A1CF
AAAS
AACS
AADAC
AADACL4
AADAT
AAGAB
AAK1
AAMDC
AAMP
uniq: write error: Broken pipe


In [21]:
!awk -F "\t" '{print $4}' human_orthos_2018.12.06.txt | sed '1,2d' | sort | uniq > human_genes_with_fish_homologues.txt


Make grep pattern more specific with print

In [22]:
!cat human_genes_with_fish_homologues.txt|awk '{print "GENEINFO="$0":"}' | head

GENEINFO=A1CF:
GENEINFO=AAAS:
GENEINFO=AACS:
GENEINFO=AADAC:
GENEINFO=AADACL4:
GENEINFO=AADAT:
GENEINFO=AAGAB:
GENEINFO=AAK1:
GENEINFO=AAMDC:
GENEINFO=AAMP:
awk: write failure (Broken pipe)
awk: close failed on file /dev/stdout (Broken pipe)
cat: write error: Broken pipe


In [23]:
!cat human_genes_with_fish_homologues.txt|awk '{print "GENEINFO="$0":"}' | xargs -I {}  grep {} humanCtoTvariant_coding.vcf > humanCtoTvariant_coding_homologues.vcf

In [25]:
!head -4 humanCtoTvariant_coding_homologues.vcf

12	53307533	309718	C	T	.	.	AF_ESP=0.0074;AF_EXAC=0.00235;AF_TGP=0.0124;ALLELEID=317931;CLNDISDB=MedGen:C0271742,OMIM:231550,Orphanet:ORPHA869,SNOMED_CT:45414006;CLNDN=Glucocorticoid_deficiency_with_achalasia;CLNHGVS=NC_000012.12:g.53307533C>T;CLNREVSTAT=criteria_provided,_single_submitter;CLNSIG=Likely_benign;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;CLNVI=Illumina_Clinical_Services_Laboratory,Illumina:797541;GENEINFO=AAAS:8086;MC=SO:0001583|missense_variant;ORIGIN=1;RS=34451260
12	53308082	309726	C	T	.	.	AF_ESP=0.0005;AF_EXAC=0.00063;AF_TGP=0.0002;ALLELEID=325873;CLNDISDB=MedGen:C0271742,OMIM:231550,Orphanet:ORPHA869,SNOMED_CT:45414006;CLNDN=Glucocorticoid_deficiency_with_achalasia;CLNHGVS=NC_000012.12:g.53308082C>T;CLNREVSTAT=criteria_provided,_single_submitter;CLNSIG=Uncertain_significance;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;CLNVI=Illumina_Clinical_Services_Laboratory,Illumina:112077;GENEINFO=AAAS:8086;MC=SO:0001583|missense_variant;ORIGIN=1;RS=112579822
12	5

In [26]:
!tail -2 humanCtoTvariant_coding_homologues.vcf

5	61544156	155772	C	T	.	.	ALLELEID=165524;CLNDISDB=MeSH:D030342,MedGen:C0950123|MedGen:C1863616,OMIM:603671|MedGen:CN517202;CLNDN=Inborn_genetic_diseases|Acromelic_frontonasal_dysostosis|not_provided;CLNHGVS=NC_000005.10:g.61544156C>T;CLNREVSTAT=criteria_provided,_multiple_submitters,_no_conflicts;CLNSIG=Pathogenic/Likely_pathogenic;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;CLNVI=OMIM_Allelic_Variant:615951.0001|UniProtKB_(protein):Q9HCJ5#VAR_071802;GENEINFO=ZSWIM6:57688;MC=SO:0001583|missense_variant;ORIGIN=33;RS=587777695
5	61530120	585031	G	A	.	.	ALLELEID=576117;CLNDISDB=MedGen:C1863616,OMIM:603671;CLNDN=Acromelic_frontonasal_dysostosis;CLNHGVS=NC_000005.10:g.61530120G>A;CLNREVSTAT=no_assertion_provided;CLNSIG=not_provided;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;GENEINFO=ZSWIM6:57688;MC=SO:0001583|missense_variant;ORIGIN=8;RS=140760439


In [27]:
cat humanCtoTvariant_coding_homologues.vcf | wc -l

86046


Count the total number of genes in original variant file before sending to Ensembl Variant effect predictor

In [98]:
!cat humanCtoTvariant_coding_homologues.vcf | cut -f 8 | awk 'match($0,/GENEINFO=[A-Z0-9]*/) {print substr($0,RSTART,RLENGTH)}'| awk -F "=" '{print $2}'|sort -u | wc -l

3887


File humanCtoTvariant_coding_homologues.vcf is sent to Ensembl Variant Effect Predictor (VEP), choose Ensembl/GENCODE basic transcripts, do not find colacated variant and make no predictions. The file obtained from VEP anntotion is named as human_variant_VEP_annotation.txt. This file include non coding gene annotation,so it needs to be filtered. 

Note: Since *GC* C to T transition won't cause stop_gained mutation, so only choose missense_variant for further analysis

In [33]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | wc -l

280948


One variant may have multiple annotation due to multiple transcript annotation in Variant Effect Predictor

In [35]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | wc -l

75928


In [37]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | head -3




awk: write failure (Broken pipe)
awk: close failed on file /dev/stdout (Broken pipe)
grep: write error: Broken pipe


Only select the genes that can complete C to T transition. As a result, only several amino acid mutation will be allowed in the final output. These mutations include "A/T","C/Y","S/N","R/H","G/D".

In [99]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | cut -f 2,6,7,9,16,17,18,19 |grep "A/T\|C/Y\|S/N\|R/H\|G/D" | head

16:70268356-70268356	AARS	ENSG00000090861	ENST00000261772	986	329	R/H	cGc/cAc
16:70269756-70269756	AARS	ENSG00000090861	ENST00000261772	824	275	G/D	gGt/gAt
2:214982225-214982225	ABCA12	ENSG00000144452	ENST00000272895	4541	1514	R/H	cGc/cAc
1:94000866-94000866	ABCA4	ENSG00000198691	ENST00000370225	6449	2150	C/Y	tGt/tAt
1:94000878-94000878	ABCA4	ENSG00000198691	ENST00000370225	6437	2146	G/D	gGc/gAc
1:94001068-94001068	ABCA4	ENSG00000198691	ENST00000370225	6320	2107	R/H	cGc/cAc
1:94010821-94010821	ABCA4	ENSG00000198691	ENST00000370225	5693	1898	R/H	cGc/cAc
1:94019691-94019691	ABCA4	ENSG00000198691	ENST00000370225	5087	1696	S/N	aGc/aAc
1:94021349-94021349	ABCA4	ENSG00000198691	ENST00000370225	4909	1637	A/T	Gcc/Acc
1:94029515-94029515	ABCA4	ENSG00000198691	ENST00000370225	4469	1490	C/Y	tGc/tAc
grep: write error: Broken pipe
cut: write error: Broken pipe


In [100]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | cut -f 2,6,7,9,16,17,18,19 |grep "A/T\|C/Y\|S/N\|R/H\|G/D"  > human_variant_VEP_extract_info.txt

In [101]:
!cut -f 2 human_variant_VEP_extract_info.txt | sort -u | wc -l

2401


In [109]:
!cut -f 2 human_variant_VEP_extract_info.txt | sort -u > human_candidate_genes.txt

Whether all 2041 gene in human_fish_orthologues.txt?

Find human fish orthologues with two methods

In [71]:
!cut -f 2,3,4,5 human_orthos_2018.12.06.txt |sed '1,2d'| sort | uniq > zebrafish_human_orthologues.txt

In [72]:
!head zebrafish_human_orthologues.txt

a1cf	apobec1 complementation factor	A1CF	APOBEC1 complementation factor
aaas	achalasia, adrenocortical insufficiency, alacrimia	AAAS	aladin WD repeat nucleoporin
aacs	acetoacetyl-CoA synthetase	AACS	acetoacetyl-CoA synthetase
aadac	arylacetamide deacetylase	AADAC	arylacetamide deacetylase
aadacl4	arylacetamide deacetylase-like 4	AADACL4	arylacetamide deacetylase like 4
aadat	aminoadipate aminotransferase	AADAT	aminoadipate aminotransferase
aagab	alpha and gamma adaptin binding protein	AAGAB	alpha and gamma adaptin binding protein
aak1a	AP2 associated kinase 1a	AAK1	AP2 associated kinase 1
aak1b	AP2 associated kinase 1b	AAK1	AP2 associated kinase 1
aamdc	adipogenesis associated, Mth938 domain containing	AAMDC	adipogenesis associated Mth938 domain containing


This dataset *zebrafish_human_orthologues.txt* have one human gene correspond to multiple zebrafish genes because the duplication of zebrafish genomes.

In [73]:
cat zebrafish_human_orthologues.txt | wc -l

16376


In [102]:
!python find_genes_for_human_fish.py human_variant_VEP_extract_info.txt zebrafish_human_orthologues.txt zebrafish_human_common_set.txt

only 2350 genes for further analysis. Compared to 2401, 51 genes are removed due to unknown reasons, waiting to figure out

In [103]:
!cat zebrafish_human_common_set.txt|wc -l

2350


In [104]:
!head zebrafish_human_common_set.txt

aanat1	arylalkylamine N-acetyltransferase 1	AANAT	aralkylamine N-acetyltransferase
aars2	alanyl-tRNA synthetase 2, mitochondrial (putative)	AARS2	alanyl-tRNA synthetase 2, mitochondrial
aars	alanyl-tRNA synthetase	AARS	alanyl-tRNA synthetase
abat	4-aminobutyrate aminotransferase	ABAT	4-aminobutyrate aminotransferase
abca12	ATP-binding cassette, sub-family A (ABC1), member 12	ABCA12	ATP binding cassette subfamily A member 12
abca1a	ATP-binding cassette, sub-family A (ABC1), member 1A	ABCA1	ATP binding cassette subfamily A member 1
abca3a	ATP-binding cassette, sub-family A (ABC1), member 3a	ABCA3	ATP binding cassette subfamily A member 3
abca4a	ATP-binding cassette, sub-family A (ABC1), member 4a	ABCA4	ATP binding cassette subfamily A member 4
abcb11a	ATP-binding cassette, sub-family B (MDR/TAP), member 11a	ABCB11	ATP binding cassette subfamily B member 11
abcb4	ATP-binding cassette, sub-family B (MDR/TAP), member 4	ABCB1	ATP binding cassette subfamily B member 1


In [107]:
!cut -f 1 zebrafish_human_common_set.txt |sort > zebrafish_selected_genes.txt

In [108]:
!cut -f 3 zebrafish_human_common_set.txt |sort > human_selected_genes.txt

In [111]:
!comm -23 human_candidate_genes.txt human_selected_genes.txt

ABCB4
AC055811.2
AC073283.3
ADGRL1
AP002884.3
ARSH
C2
C3
CEL
CLCNKB
CNGB3
DNAI1
DPYD
DSG2
F13A1
FUCA1
G6PC
GJA1
GJB1
GJB2
GJB3
GJB4
GJC2
GPR82
HARS
HCCS
IL21R
ITGA2
KCNJ1
KIAA0586
KRT18
KRTAP10-10
LIM2
LRTOMT
MATR3
MUT
NKX2-1
NKX2-5
OGT
PIEZO2
PRCD
SFTPC
SLC26A3
SLC6A19
SLC9A3
SP110
TAPBP
TPMT
TTN
TXNRD2
WISP3


In [112]:
!grep WISP3 zebrafish_human_orthologues.txt

In [113]:
! grep WISP3 human_genes_with_fish_homologues.txt

In [114]:
!grep WISP3 human_orthos_2018.12.06.txt

It seems like that some genes like WISP3 was added to 2041 gene list by Ensembl VEP annotation

In [116]:
!grep TTN human_orthos_2018.12.06.txt | head -3

ZDB-GENE-030113-2	ttn.2	titin, tandem duplicate 2	TTN	titin	188840	7273	12403	AA	ZDB-PUB-040609-1
ZDB-GENE-030113-2	ttn.2	titin, tandem duplicate 2	TTN	titin	188840	7273	12403	AA	ZDB-PUB-050823-6
ZDB-GENE-030113-2	ttn.2	titin, tandem duplicate 2	TTN	titin	188840	7273	12403	AA	ZDB-PUB-070330-28


In [117]:
!grep TTN human_genes_with_fish_homologues.txt

CTTN
CTTNBP2
CTTNBP2NL
RTTN
TTN


In [118]:
!grep TTN zebrafish_human_orthologues.txt

cttnbp2	cortactin binding protein 2	CTTNBP2	cortactin binding protein 2
cttnbp2nla	CTTNBP2 N-terminal like a	CTTNBP2NL	CTTNBP2 N-terminal like
cttnbp2nlb	CTTNBP2 N-terminal like b	CTTNBP2NL	CTTNBP2 N-terminal like
cttn	cortactin	CTTN	cortactin
rttn	rotatin	RTTN	rotatin
ttn.1	titin, tandem duplicate 1	TTN	titin
ttn.2	titin, tandem duplicate 2	TTN	titin


Some gene are excluded from 2014 gene list because their weird zebrafish gene name

12/25/2018 some data didn't save correctly

workflow: final_candidates.py selected genes dataset were sent to ensembl and download the gene_transcript_id and protein sequence datafile, the use find_conserved_transcript.py to get the conserved transcript. Then use the python find_chromo_pos.py script to get the final output for further analysis.

12/26/2018 analyze the positive strand

In [1]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | cut -f 2,6,7,9,16,17,18,19 |grep "A/V" | head

9:104822514-104822514	ABCA1	ENSG00000165029	ENST00000374736	2810	937	A/V	gCg/gTg
2:215019657-215019657	ABCA12	ENSG00000144452	ENST00000272895	1427	476	A/V	gCg/gTg
1:93997943-93997943	ABCA4	ENSG00000198691	ENST00000370225	6647	2216	A/V	gCg/gTg
1:94043413-94043413	ABCA4	ENSG00000198691	ENST00000370225	3113	1038	A/V	gCc/gTc
1:94079341-94079341	ABCA4	ENSG00000198691	ENST00000370225	1220	407	A/V	gCa/gTa
1:94111561-94111561	ABCA4	ENSG00000198691	ENST00000370225	179	60	A/V	gCg/gTg
2:168944621-168944621	ABCB11	ENSG00000073734	ENST00000263817	2594	865	A/V	gCt/gTt
12:21908099-21908099	ABCC9	ENSG00000069431	ENST00000261200	1433	478	A/V	gCa/gTa
12:120737939-120737939	ACADS	ENSG00000122971	ENST00000242592	575	192	A/V	gCt/gTt
12:51920826-51920826	ACVRL1	ENSG00000139567	ENST00000388922	1445	482	A/V	gCg/gTg
grep: write error: Broken pipe
cut: write error: Broken pipe


In [2]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | cut -f 2,6,7,9,16,17,18,19 |grep "A/V" | wc -l

4537


In [3]:
!grep -E 'missense_variant' human_variant_VEP_annotation.txt | awk '$1!=prev {print $0} {prev=$1}' | cut -f 2,6,7,9,16,17,18,19 |grep "A/V" > human_variant_VEP_extract_info_positive.txt

In [4]:
!python find_genes_for_human_fish.py human_variant_VEP_extract_info_positive.txt zebrafish_human_orthologues.txt zebrafish_human_common_set_positive.txt

In [5]:
!head zebrafish_human_common_set_positive.txt

aars2	alanyl-tRNA synthetase 2, mitochondrial (putative)	AARS2	alanyl-tRNA synthetase 2, mitochondrial
aars	alanyl-tRNA synthetase	AARS	alanyl-tRNA synthetase
abat	4-aminobutyrate aminotransferase	ABAT	4-aminobutyrate aminotransferase
abca12	ATP-binding cassette, sub-family A (ABC1), member 12	ABCA12	ATP binding cassette subfamily A member 12
abca1a	ATP-binding cassette, sub-family A (ABC1), member 1A	ABCA1	ATP binding cassette subfamily A member 1
abca4a	ATP-binding cassette, sub-family A (ABC1), member 4a	ABCA4	ATP binding cassette subfamily A member 4
abcb11a	ATP-binding cassette, sub-family B (MDR/TAP), member 11a	ABCB11	ATP binding cassette subfamily B member 11
abcb7	ATP-binding cassette, sub-family B (MDR/TAP), member 7	ABCB7	ATP binding cassette subfamily B member 7
abcc6a	ATP-binding cassette, sub-family C (CFTR/MRP), member 6a	ABCC6	ATP binding cassette subfamily C member 6
abcc8	ATP-binding cassette, sub-family C (CFTR/MRP), member 8	ABCC8	ATP binding cassette subfamily C me

In [6]:
!cat zebrafish_human_common_set_positive.txt |wc -l

1447


In [8]:
!cut -f 1 zebrafish_human_common_set.txt |sort > zebrafish_selected_genes_positive.txt
!cut -f 3 zebrafish_human_common_set.txt |sort > human_selected_genes_positive.txt

In [10]:
#final_candidates.py
f1 = open("zebrafish_selected_genes_positive.txt","r")
f2 = open("zebrafish_human_common_set.txt","r")
f3 = open("zebrafish_id_conversion_positive.txt","r")
f4 = open("human_id_conversion_positive.txt","r")

fishgenes = []
for line in f1.readlines():
	fishgene = line.strip()
	fishgenes.append(fishgene)
print len(fishgenes)
print fishgenes[:10]

humangenes = []
genesPair = []
for line in f2.readlines():
	fishgene = line.strip().split("\t")[0]
	humangene = line.strip().split("\t")[2]
	if fishgene in fishgenes:
		humangenes.append(humangene)
		genesPair.append(fishgene+humangene)
		#print fishgene+"\t"+humangene
print len(humangenes)
print humangenes[:10]

fishTranscript = []
for line in f3.readlines():
	thisline = line.strip().split("\t")
	gene = thisline[2]
	transcript = thisline[1]
	if gene in fishgenes:
		fishgenes.remove(gene)
		fishTranscript.append(gene+"\t"+transcript)

print len(fishTranscript)

humanTranscript = []
for line in f4.readlines():
        thisline = line.strip().split("\t")
        gene = thisline[2]
        transcript = thisline[1]
        if gene in humangenes:
                humangenes.remove(gene)
                humanTranscript.append(gene+"\t"+transcript)


print len(humanTranscript)

f5 = open("fish_human_final_candidate_positive.txt","w")
for fishGeneTranscript in fishTranscript:
	fishgene = fishGeneTranscript.split("\t")[0]
	for humanGeneTranscript in humanTranscript:
		humangene = humanGeneTranscript.split("\t")[0]
		fishHuman = fishgene+humangene
		if fishHuman in genesPair:
			f5.write(fishGeneTranscript+"\t"+humanGeneTranscript+"\n")

f1.close()
f2.close()
f3.close()
f4.close()
f5.close()

2350
['aanat1', 'aars', 'aars2', 'abat', 'abca12', 'abca1a', 'abca3a', 'abca4a', 'abcb11a', 'abcb4']
2350
['AANAT', 'AARS2', 'AARS', 'ABAT', 'ABCA12', 'ABCA1', 'ABCA3', 'ABCA4', 'ABCB11', 'ABCB1']
2248
2350


In [11]:
!head -3 fish_human_final_candidate_positive.txt

gria3a	ENSDART00000137120	GRIA3	ENST00000616590
adamts13	ENSDART00000146364	ADAMTS13	ENST00000628339
gnal	ENSDART00000140924	GNAL	ENST00000334049


In [12]:
!cat fish_human_final_candidate_positive.txt | wc -l

2248


In [14]:
!cut -f 2 fish_human_final_candidate_positive.txt > zebrafish_selected_transcripts_positive.txt
!cut -f 4 fish_human_final_candidate_positive.txt >  human_selected_transcripts_positive.txt

In [17]:
!python find_conserved_transcript_positive.py

In [18]:
!head human_zebrafish_conserved_variant_positive.txt

OTC	209	otc	207	A	A
OTC	233	otc	231	A	A
MYO15A	1772	myo15aa	2382	A	A
MYO15A	2248	myo15aa	2857	A	A
DLD	61	dldh	60	A	A
DLD	241	dldh	239	A	A
ABCC6	594	abcc6a	562	A	A
ABCC6	187	abcc6a	158	A	A
TSC2	862	tsc2	694	A	A
TSC2	289	tsc2	122	A	A


In [19]:
!cat human_zebrafish_conserved_variant_positive.txt|wc -l

658


In [ ]:
!python find_chromo_pos_positive.py

In [20]:
!head all_variants_positive.txt

MYO15A	ENST00000205890	myo15aa	ENSDART00000149546	2382	A/V	AGAGGGATTGCTGAAATTTGGcCGCCACCGTTGAggCCTGAT
DLD	ENST00000205402	dldh	ENSDART00000006709	60	A/V	GTACCTTAAACCCAAGCTGAGcTGCTTTGATAGCggCAACAT
DLD	ENST00000205402	dldh	ENSDART00000006709	239	A/V	CATGGCCAAGAAATTCCACTGcTGTAACTTTggCACCCAACC
ABCC6	ENST00000205557	abcc6a	ENSDART00000189619	158	A/V	ACTCTTCCTCAGCTGTTTTGcAGACCAggCACCCTTAggCAA
TSC2	ENST00000219476	tsc2	ENSDART00000158948	122	A/V	TAGGGTGTACTCCGAGGACGcCGCTCTCCTCCGCggTGCTGT
TSC2	ENST00000219476	tsc2	ENSDART00000158948	264	A/V	CACGCTGGTCTCCTACAGAGcTCAggCCATCCAGCCggCCAA
TSC2	ENST00000219476	tsc2	ENSDART00000158948	454	A/V	TTTCCTGCTGCTGATGAGGGcTGATTCTCTACATCGTCTCgg
TSC2	ENST00000219476	tsc2	ENSDART00000158948	597	A/V	CCTCACTGACGTTCAGCTCGcTGTggTTCCTGTTCTCACggC
TSC2	ENST00000219476	tsc2	ENSDART00000158948	726	A/V	TCTGGCTCATCATGTGATCGcCATGTggTTCATCCGCTGCCG
TSC2	ENST00000219476	tsc2	ENSDART00000158948	1041	A/V	CATGCTGACGCAGGGCTGGGcCGAGATCTTCATCCGCAggCC


In [21]:
!cat all_variants_positive.txt | wc -l

369
